osquery/remote/enroll/plugins/tls.cpp

```diff
@@ -20,6 +22,12 @@
 #include "osquery/core/process.h"
 
 namespace osquery {
+namespace tables {
+QueryData genOsqueryInfo(QueryContext& context);
```

Comment:
The main consideration here was making assumptions about how much of the system can be expected to be initialized at the time this code is likely to be called. If the "config_tls_enroll_on_startup" option is set, then enroll gets called at the time that the active config plugin gets initialized (osquery/core/init.cpp, line~587). This is *before* the Registry::setup() is called. I must confess that I do not have knowledge of the designers' full (future) intentions here, but I was driven by the comment "Run the setup for all lazy registries (tables, SQL)" describing the call to Registry::setUp() to be defensive and chose to call the genXXX queries directly.

Happy to switch to using the SQL class